server/services/nl.py

```diff
@@ -29,6 +32,205 @@
+
+  buckets = {}
+  for rpv in cpv_parts:
+    parts = [fixed_prefix]
+    v_only = ""
+    for opv in cpv_parts:
+      if rpv == opv:
+        # P only
+        p_only, v_only = rpv.split('=')
+        parts.append(p_only + '=')
+      else:
+        parts.append(opv)
+    buckets[','.join(parts)] = v_only
+  return buckets
+
+def _chart_config(place_dcid, main_place_type, main_place_name, child_places_type, 
+                  highlight_svs, sv2name, fixed_props, peer_buckets):
+  #@title
+  chart_config = {}
+  chart_config['metadata'] = {
+    'place_dcid': [place_dcid],
+  }
+
+  if child_places_type:
+    chart_config['metadata']['contained_in_places'] = [{
+      'key': main_place_type,
+      'value': child_places_type
+    }]
+
+  added_buckets = set()
+  sv4spec = set()
+
+  blocks = []
+  for sv in highlight_svs:
+    tiles = []
+
+    tiles.append({
+        'title': sv2name[sv] + ': historical',
+        'type': 'LINE',
+        'stat_var_key': [sv]})
+
+    if child_places_type:
+      tiles.append({
+          'title': sv2name[sv] + ': places within ' + main_place_name,
+          'type': 'MAP',
+          'stat_var_key': [sv]})
+
+      tiles.append({
+          'title': sv2name[sv] + ': rankings within ' + main_place_name,
+          'type': 'RANKING',
+          'stat_var_key': [sv],
+          'ranking_tile_spec': {
+              'show_highest': True,
+              'show_lowest': True
+          }
+      })
+
+    # If the highlight SV is part of an SV peer-group, add it ahead of others.
+    for key, svs in peer_buckets.items():
+      if sv not in svs: continue
+      if key in added_buckets: continue
+
+      sv_list = list(svs)
+      # Consider if this should be bar or time-series?
+      tiles.append({
+            'title': _bucket_to_name(key, fixed_props),
+            'type': 'LINE',
+            'stat_var_key': sv_list
+      })
+      added_buckets.add(key)
+      sv4spec.update(sv_list)
+      # TODO: add CLUSTERED_BAR
+
+    blocks.append({
+        'title': sv2name[sv],
+        'columns': [{
+            'tiles': tiles
+        }]
+    })
+    sv4spec.add(sv)
+
+  for key, svs in peer_buckets.items():
+    if key in added_buckets: continue
+    tile = {
+      'title': _bucket_to_name(key, fixed_props),
+      'type': 'BAR',
+      'stat_var_key': list(svs)
+    }
+    blocks.append({
+        'title': _bucket_to_name(key, fixed_props) + ' in ' + main_place_name,
+        'columns': [{
+            'tiles': [tile]
+        }]
+    })
+    added_buckets.add(key)
+    sv4spec.update(list(svs))
+    # TODO: add CLUSTERED_BAR when supported
+
+  sv_specs = []
+  for sv in sv4spec:
+    sv_specs.append({
+        'key': sv,
+        'value': {
+            'stat_var': sv,
+            'name': sv2name[sv]
+        }
+    })
+
+  chart_config['categories'] = [{
+      'title': 'Search Results',
+      'blocks': blocks,
+      'stat_var_spec': sv_specs
+  }]
+
+  return json.dumps(chart_config, indent=2)
```

Comment:
just return chart_config here, since this is the client library.

can do json.dumps() in /router/nl_iterface.py

Reply:
Done